From 6143775eacfdb734f392fef8b6c29c13aa35c1ea Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Thu, 13 Jun 2013 15:04:28 +0200
Subject: [PATCH 06/39] ASoC: core: soc-fw: Small simplification in
 soc_fw_dapm_graph_load()

Issue: GMINL-3679

When checking the ->control parameter for the route, the code can be made
a bit more readable.

Change-Id: I0263189311a7cc5223c66e96d7bafef1e921af17
Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
---
 sound/soc/soc-fw.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/soc-fw.c b/sound/soc/soc-fw.c
index 9dc014d..ee0fbc3 100644
--- a/sound/soc/soc-fw.c
+++ b/sound/soc/soc-fw.c
@@ -715,10 +715,10 @@ static int soc_fw_dapm_graph_load(struct soc_fw *sfw,
 
 		route.source = elem->source;
 		route.sink = elem->sink;
-		if (strnlen(elem->control, SND_SOC_FW_TEXT_SIZE) == 0)
-			route.control = NULL;
-		else
+		if (strnlen(elem->control, SND_SOC_FW_TEXT_SIZE))
 			route.control = elem->control;
+		else
+			route.control = NULL;
 
 		/* add route, but keep going if some fail */
 		snd_soc_dapm_add_routes(dapm, &route, 1);
-- 
1.9.1

